src/MockClassifier.Api/Controllers/MessagesController.cs

```diff
@@ -0,0 +1,97 @@
+using MockClassifier.Api.Services.Dmr;
+using System.Text;
+using System.Text.Json;
+
+namespace MockClassifier.Api.Controllers
+{
+    [Route("/dmr-api/messages")]
+    [ApiController]
+    public class MessagesController : ControllerBase
+    {
+        private readonly IDmrService _dmrService;
+        private readonly ITokenService _tokenService;
+        private readonly INaturalLanguageService _naturalLanguageService;
+        private readonly IEncodingService _encodingService;
+
+        public MessagesController(IDmrService dmrService, ITokenService tokenService, INaturalLanguageService naturalLanguageService, IEncodingService encodingService)
+        {
+            _dmrService = dmrService;
+            _tokenService = tokenService;
+            _naturalLanguageService = naturalLanguageService;
+            _encodingService = encodingService;
+        }
+
+        /// <summary>
+        /// Processes a string to identify classifications and issues call backs to Dmr for each classification.
+        /// </summary>
+        /// <param name="input">Property which contains the base64 encoded payload containing the message to be classified</param>
+        /// <returns>An empty 202/Accepted result</returns>
+        [HttpPost]
+        public async Task<IActionResult> Post()
+        {
+            // Get base64 encoded body as Payload
+            using StreamReader reader = new(Request.Body, Encoding.UTF8);
+            var input = await reader.ReadToEndAsync().ConfigureAwait(false);
+            if (string.IsNullOrEmpty(input))
+            {
+                return BadRequest(ModelState);
+            }
+            var decodedInput = _encodingService.DecodeBase64(input);
+            var payload = JsonSerializer.Deserialize<DmrRequestPayload>(decodedInput);
+
+            // Classify
+            List<string> classifications = _naturalLanguageService.Classify(payload.Message).ToList();
+            classifications = classifications.Concat(_tokenService.Classify(payload.Message).ToList()).ToList();
+
+            // Send Dmr call back(s)
+            foreach (var classification in classifications)
+            {
+                var dmrRequest = GetDmrRequest(payload.Message, classification, Request.Headers);
+                _dmrService.RecordRequest(dmrRequest);
+            }
+
+            return Accepted();
+        }
+
+        /// <summary>
+        /// Builds a DmrRequest object from parameters
+        /// </summary>
+        /// <param name="message">The message to go into the .Payload.Message property</param>
+        /// <param name="classification">The classification to go into the .Payload.Classification property</param>
+        /// <param name="headers">The header collection from the origional Request. Used to create the .Header object</param>
+        /// <returns>A DmrRequest object</returns>
+        private static DmrRequest GetDmrRequest(string message, string classification, IHeaderDictionary headers)
```

Comment:
Thoughts on moving this logic into another class outside of the controller? would be nice for controller to only display HTTP/endpoint/route related methods only

Reply:
Usually I would agree with you but this is just a mock service so if it's okay with you I'd like to leave this as is. Is that all right?

Reply:
Yahya and I discussed offline and we're happy to leave it as is. Thanks!